content/src/ports/activeEntities.ts

```diff
@@ -0,0 +1,241 @@
+import { Entity, EntityId, EntityType, Pointer } from 'dcl-catalyst-commons'
+import LRU from 'lru-cache'
+import { EnvironmentConfig } from '../Environment'
+import { mapDeploymentsToEntities } from '../logic/deployments'
+import { getDeployments } from '../service/deployments/deployments'
+import { AppComponents } from '../types'
+
+export const NOT_ACTIVE = 'NONE'
+export type EntityCacheResult = Entity | typeof NOT_ACTIVE
+export type PointerToEntityId = EntityId | typeof NOT_ACTIVE
+
+const isEntityPresent = (result: EntityCacheResult | undefined): result is Entity => result !== NOT_ACTIVE
+const isPointingToEntity = (result: PointerToEntityId | undefined): result is EntityId => result !== NOT_ACTIVE
+
+export type ActiveEntities = {
+  /**
+   * Retrieve active entities that are pointed by the given pointers
+   * Note: result is cached, even if the pointer has no active entity
+   */
+  withPointers(pointers: Pointer[]): Promise<Entity[]>
+  /**
+   * Retrieve active entities by their ids
+   * Note: result is cached, even if the id has no active entity
+   */
+  withIds(entityIds: EntityId[]): Promise<Entity[]>
+  /**
+   * Save entityId for given pointer and store the entity in the cache,
+   * useful to retrieve entities by pointers
+   */
+  update(pointers: Pointer[], entity: EntityCacheResult): void
+  /**
+   * Returns the cached result:
+   *  - entity id if there is an active entity
+   *  - NONE if there is no active entity
+   *  - undefined if there is no cached result
+   * Note: testing purposes
+   */
+  getCachedActiveEntityOrNone(pointer: Pointer): PointerToEntityId | undefined
```

Comment:
I understand the motivation of exhausting tools for testing this key component. In addition to exposing internal behavior of having a cache (ActiveEntities is not a cache as far as i understand), what if we inject the `cache` and `entityIdByPointers` to the creation method so then in the test we can make the assertions/mock what we want?